pkg/kubelet/kubelet.go

```diff
@@ -2139,6 +2151,8 @@ func (kl *Kubelet) HandlePodActions(update kubetypes.PodUpdate) {
 func (kl *Kubelet) HandlePodAdditions(pods []*v1.Pod) {
 	start := kl.clock.Now()
 	sort.Sort(sliceutils.PodsByCreationTime(pods))
+	kl.podResizeMutex.Lock()
```

Comment:
for best practice, this function scope lock should be avoid as possible.

Reply:
this is necessary. when kubelet restarts, all the pods for the node are sent by apiserver. and we need to process them in the lock so they get re-added in the order they were initially added. if we do this on per pod basis instead of the entire sequence, there is a chance that a pending resize of an earlier pod would be accepted before all the original pods could be added, resulting in a pod at the end of the list that was successfully added being rejected on kubelet restart.